### DISCUSSION TITLE
20240919 Weekly Development Meeting

### DISCUSSION THREAD

**Discussion by josecelano** [nautilus-cyberneering/.github#108] [Meetings]
- Time:  9:30 in Canary Islands (WET (UTC+0)).
- Time: 10:30 in Vaduz (CET (UTC+1)).

Location:
<https://meeting.blockfinance-eco.li/NautilusCyberneeringWeeklyDevMeeting>

Expected duration: 2 to 4 hours with breaks

Communicated attendants:

- Cameron Garnham(@da2ce7)  - x 
- Jose Celano(@josecelano) - x
- Constantin Bosse(@cgbosse) - x
- Mario Melchor(@mario-nt) - x
- Graeme Byrne(@grmbyrn) - x

-------------------------------------------------------------

## Agenda:

1. Everyone will take their turn and:
- Go over his notes added as comments for the meeting which will have been added in advance to give an update of all the work done during the previous week indicating.
	- The issues worked on
	- The progress made on them
	- Other things done
- Inform of any faced challenges and request help if necessary to work on specific issues.
- Highlight the encountered difficulties, especially if there is a risk of them causing delays or affecting end product's quality.
- Communicate what will be the next week's tasks.
- List any new topics for further discussion in an additional Technical meeting that he noted for future discussion in comments to this discussion either before the meeting or after his turn so that they get considered.

During the meeting every team member will take their turns and follow this sequence.
The topics for further discussion will be noted in advance by each the team member and to the meetings agenda.

2. At the end of the meeting if there any new technical topics to be discussed further it will be decided to
set another meeting date for the discussion of these or to hold the meeting hereafter.

Also the persons to be part of the meeting will be decided on so that the others will be free to continue their own work.

**Comment by josecelano** [comment]
### Torrust Index and GUI

Issues:

- https://github.com/torrust/torrust-index-gui/issues/614
  - https://github.com/torrust/torrust-index-gui/issues/615
  - https://github.com/torrust/torrust-index-gui/issues/617
  - https://github.com/torrust/torrust-index-gui/issues/619
  - https://github.com/torrust/torrust-index-gui/issues/622

These new legal texts have also been deployed to the [demo site](https://index.torrust-demo.com/).

### Torrust Tracker

- https://github.com/torrust/torrust-tracker/issues/675

On this [PR](https://github.com/torrust/torrust-tracker/pull/1041), you can see all the options you have now to pass UDP and HTTP tracker URLs to the Tracker Checker.

### Torrust Bencode2Json

Some of the [Tracker Checker issues](https://github.com/torrust/torrust-tracker/issues/669) require conversion from bencode format to JSON.

I've been [researching for packages](https://github.com/torrust/teps/discussions/15#discussioncomment-10611909) to use. See [here](https://github.com/torrust/teps/discussions/15#discussioncomment-10611909).

Finally, I decided to create a new one based on [this C implementation](https://gist.github.com/camilleoudot/840929699392b3d25afbec25d850c94a). I'm working on it. The implementation is almost finished (not published yet):

- [x] Integer
- [x] String
- [x] List
- [ ] Dictionary **WIP**

I have to finish the implementation and then clean it (the current implementation has a lot of duplicated code and bad names).

I decided to use this version because:

- It's fast
- It doesn't consume much memory
- It can handle large files (not yet; it's storing the final JSON is a string, but it can easily be changed to print it on the fly)

However, it's not a direct port from the C version. I want to:

- Remove magic numbers to make the code more readable.
- Add tests.
- Return good errors (concrete, context, position) with the exact reason why the encoded value is not well-formatted and the position (byte) when the parsing failed.
- Print non UTF-8 strings as hex bytes (`<hex>ff</hex>`) like https://chocobo1.github.io/bencode_online/.

**Comment by mario-nt** [comment]
### Index
	
- https://github.com/torrust/torrust-index/pull/718

> - [x] fix clippy errors
> - [x] 1 to 1 with Jose
> - [ ] Fix failing tests with private tracker


 	



### Other

- Setting up Docker
- Runing E2E tests with Docker 101
